onnxruntime/core/providers/cpu/ml/svmclassifier.h

```diff
@@ -31,28 +31,28 @@ class SVMCommon {
 
   float kernel_dot(const T* A, int64_t a, const std::vector<float>& B, int64_t b, int64_t len, KERNEL k) const {
     float sum = 0.f;
+    float* pA = (float*)&A[a];
+    float* pB = (float*)&B[b];
     if (k == KERNEL::POLY) {
-      for (int64_t i = 0; i < len; i++) {
-        sum += B[b + i] * static_cast<float>(A[a + i]);
-      }
+      for (int64_t i = len; i > 0; --i, ++pA, ++pB)
```

Comment:
>++pA, ++pB [](start = 40, length = 10)

pA += sizeof(T)

Reply:
Why pA += sizeof(T)? pA++ is equivalent to ((char*)pA) + sizeof(T)?

Reply:
My bad. I intended to say the previous setting here only works with `float`.